### PR TITLE
[Trainer] Add embedding trainer

### PR DESCRIPTION
**Pull request by DesmonDay**
### PR types
New features

### PR changes
Others

### Description
Add embedding trainer.

### CODE REVIEW

File: paddlenlp/transformers/embedding_utils.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+from paddle.distributed import fleet
+
+
+def dist_gather_tensor_with_gradient(tensor):
```
Comment (ZHUI) [inline paddlenlp/transformers/embedding_utils.py:19]: 可以找其他 utils 的目录放吧。感觉还比较通用。gathe across dp

函数里面的 _gather_tensor_with_gradient 这个 gradient怎么体现的？

File: paddlenlp/trl/embedding_trainer.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+from paddlenlp.transformers.embedding_utils import dist_gather_tensor_with_gradient
+
+__all__ = ["EmbeddingTrainer"]
+
+
+class EmbeddingTrainer(Trainer):
+    def __init__(self, model_args, **kwargs):
+        super().__init__(**kwargs)
+
+        self.model_args = model_args
```
Comment (ZHUI) [inline paddlenlp/trl/embedding_trainer.py:35]: 最好加一个 loss type之类的字段。我看zhangjie那边还需要加inf-cl的loss

**Review by ZHUI — APPROVED** [review]
LGTM

File: paddlenlp/transformers/embedding_utils.py (hunk 1 above)
Comment (DesmonDay) [reply paddlenlp/transformers/embedding_utils.py:19]: 体现在这里，重新把tensor赋值进去。all_tensors[sharding_rank] = tensor
